### PR TITLE
Feature: wire api-server to sdk-services

### PR DESCRIPTION
**Pull request by gsora**
- [x] update `demeris-backend-models` to latest `HEAD` containing `cosmos_sdk_version`  in the `cns.Chain` struct
- [x] replace anything concerning Cosmos SDK with the analogous concept from `sdk-service-meta`

### CODE REVIEW

**Comment by gsora** [comment]
Ok so, I ported over every substantial SDK GRPC query to `sdk-service` queries, essentially removing the need for `codec.Marshaler` all over the `api` package.

`api-server` still depends on Cosmos SDK in two major clusters of code:
 - `api`:
   -  `relayer` include SDK types and bech32 functions, because this method has to compare balances
   - `liquidity` which includes SDK just for the `Coins` type, for which we could provide a general implementation since it's only used to format caller return data
  - `store`, which is still tightly coupled to SDK but which should be extracted under the `utils` package anyway

I would *really* like to throw those last two dependencies away to make sure we're completely decoupled from Cosmos SDK, which translates to quick build times and overall higher code quality.

How do we wanna go about doing this?

Paging @sgerogia @sahith-narahari and @DeshErBojhaa.

**Comment by sgerogia** [comment]
An uninformed suggestion, due to my cursory knowledge of Go and SDK.

Could the different library versions be wrapped in a `utils` abstraction layer with methods returning generic types?
I.e. 
* import all different SDK package versions in utils using [replace](https://www.percona.com/blog/2020/03/09/using-different-versions-of-a-package-in-an-application-via-go-modules/)
* move all parsing/conversion methods to `utils` (e.g. `enoughBalance`), passing the new `chain.version` param
* do stuff and return the generic `utils` type (e.g. `utils.Coins`)

**Comment by gsora** [comment]
> import all different SDK package versions in utils using replace

Sadly we cannot do that, because of how the Go `mod` system works and how the Cosmos SDK is versioned

> - move all parsing/conversion methods to utils (e.g. enoughBalance), passing the new chain.version param
> - do stuff and return the generic utils type (e.g. utils.Coins)

The issue here is that regardless of what we choose, we might end up having a Cosmos SDK dependency anyway especially since we rely on `types.Coins` comparison, which honestly I wouldn't end up reimplementing from scratch.

We could import that code in our `utils` package (copy-pasting it and documenting), or keep a dependency in `api-server` to the SDK, documenting the fact that it should only be used for *very* common and stable types.

File: api/account/api.go [hunk 1]
```diff
@@ -286,122 +275,91 @@ func GetDelegatorRewards(c *gin.Context) {
...
-	addressBytes, err := hex.DecodeString(address)
-
+	client, err := sdkservice.Client(chain.MajorSDKVersion())
 	if err != nil {
 		e := deps.NewError(
-			"delegator rewards",
-			fmt.Errorf("unable to decode hex to bytes"),
+			"chains",
+			fmt.Errorf("cannot retrieve sdk-service for version %s with chain name %v", chain.CosmosSDKVersion, chain.ChainName),
 			http.StatusBadRequest,
```
Comment (sgerogia) [inline api/account/api.go]: Is this a 400 or 500?
I.e. everything is correctly configured and still the caller can cause an error?

Comment (gsora) [reply api/account/api.go]: Hmm, looking at it, it might be more sensible to return a 500 here.

Comment (sgerogia) [reply api/account/api.go]: There are 1-2 other places where this also occurs

**Comment by sgerogia** [comment]
I will now utter the 2 dreaded words: unit tests?

**Comment by gsora** [comment]
> I will now utter the 2 dreaded words: unit tests?

Considering the state of the codebase and the scope of this task, I'd say it's better we leave unit testing alone until we have an architecture in place.

**Comment by gsora** [comment]
@sgerogia fixed HTTP status codes for sdk-service-related tests, I'm gonna try and fix the CI then I'll merge.

**Comment by gsora** [comment]
CI is happy, merging!

**Pull request merged**